code/game/machinery/autolathe.dm

```diff
@@ -145,88 +145,115 @@
+		if(istext(material)) // category
+			var/list/choices = list()
+			for(var/datum/material/valid_candidate as anything in SSmaterials.materials_by_category[material])
+				if(materials.get_material_amount(valid_candidate) < amount_needed)
+					continue
+				choices[valid_candidate.name] = valid_candidate
+			if(!length(choices))
+				say("No valid materials with applicable amounts detected for design.")
+				return
+			var/chosen = tgui_input_list(
+				ui.user,
+				"Select the material to use",
+				"Material Selection",
+				sort_list(choices),
+			)
+			if(isnull(chosen))
+				return // user cancelled
+			material = choices[chosen]
+
+		if(isnull(material))
+			stack_trace("got passed an invalid material id: [material]")
 			return
+		total_materials_used[material] = amount_needed * build_count
+		materials_per_item[material] = amount_needed
 
-		//use power
-		var/total_amount = 0
-		for(var/material in being_built.materials)
-			total_amount += being_built.materials[material]
-		use_power(max(active_power_usage, (total_amount) * multiplier / 5))
+	if(!materials.has_materials(total_materials_used))
+		say("Not enough materials to begin production.")
+		return
 
-		//use materials
-		materials.use_materials(materials_used, coeff, multiplier)
-		to_chat(usr, span_notice("You print [multiplier] item(s) from the [src]"))
-		update_static_data_for_all_viewers()
+	var/power_use_amount = 0
+	for(var/material_used in total_materials_used)
+		power_use_amount += total_materials_used[material_used] * 0.2
+	if(!directly_use_power(power_use_amount))
+		say("Not enough power in local network to begin production.")
+		return
 
-		//print item
-		icon_state = "autolathe_n"
-		var/time_per_item = is_stack ? 32 : ((32 * coeff * multiplier) ** 0.8) / multiplier
-		make_items(custom_materials, multiplier, is_stack, usr, time_per_item)
+	var/build_time_per_item = design.construction_time * creation_efficiency * design.lathe_time_factor
+	materials.use_materials(total_materials_used)
+	start_making(design, build_count, ui.user, build_time_per_item, materials_per_item)
+	return TRUE
 
-		return TRUE
+/obj/machinery/autolathe/proc/start_making(datum/design/design, build_count, mob/user, build_time_per_item, list/materials_per_item)
+	busy = TRUE
+	icon_state = "autolathe_n"
+	update_static_data_for_all_viewers()
+
+	for(var/idx in 1 to build_count)
+		addtimer(CALLBACK(src, PROC_REF(do_make_item), design, materials_per_item), build_time_per_item * idx)
+	addtimer(CALLBACK(src, PROC_REF(finalize_build)), (build_time_per_item * build_count) + 1)
+
+/obj/machinery/autolathe/proc/do_make_item(datum/design/design, list/materials_per_item)
+	var/turf/target = get_step(src, drop_direction)
+	if(isclosedturf(target))
+		target = get_turf(src)
+	var/atom/movable/created = new design.build_path(design.build_path)
+	created.set_custom_materials(materials_per_item.Copy())
+	created.forceMove(target)
+
+/obj/machinery/autolathe/proc/finalize_build()
```

Comment:
This proc is just 2 lines. You could easily move it's code to where it's required or autodoc it, maybe make it PRIVATE(TRUE) but that's your call